client/components/Item/MediaStreamSelector.vue

```diff
@@ -0,0 +1,210 @@
+       */
+      get(): { text: SelectItems; value: number | undefined }[] {
+        const items = this.mediaStreams.map((value, _idx) => {
+          return {
+            text: {
+              selection: this.getTrackSelection(value),
+              subtitle: this.getTrackSubtitle(value),
+              icon: this.getTrackIcon(value),
+              title: this.getTrackTitle(value)
+            },
+            value: value.Index
+          };
+        });
+
+        items.unshift({
+          value: -1,
+          text: {
+            selection: this.$t('disabled'),
+            title: this.$t('disabled'),
+            subtitle: undefined,
+            icon: undefined
+          }
+        });
+
+        return items;
+      }
+    },
+    /**
+     * @returns {number|undefined} Default index to use (undefined if none)
+     */
+    defaultIndex: {
+      get(): number | undefined {
+        if (this.defaultStreamIndex !== undefined) {
+          return this.defaultStreamIndex;
+        }
+
+        return this.mediaStreams.find((track) => track.IsDefault)?.Index;
+      }
+    }
+  },
+  /**
+   * Sets the default track when loading the component
+   */
+  beforeMount() {
+    if (this.defaultIndex !== undefined) {
+      this.trackIndex = this.defaultIndex;
+    }
+
+    this.$emit('input', this.trackIndex);
+  },
+  methods: {
+    /**
+     * @param {MediaStream} track - Track to parse
+     * @returns {string} Text to display in select when track is choosen
+     */
+    getTrackSelection(track: MediaStream): string {
+      if (track.DisplayTitle) {
+        return track.DisplayTitle;
+      }
+
+      return '';
+    },
+    /**
+     * @param {MediaStream} track - Track to parse
+     * @returns {string|undefined} Optional icon to use for the track line in the v-select menu
+     */
+    getTrackIcon(track: MediaStream): string | undefined {
+      if (this.type === 'Audio' && track.ChannelLayout) {
+        return this.getSurroundIcon(track.ChannelLayout);
+      }
+
+      return undefined;
+    },
+    /**
+     * @param {MediaStream} track - Track to parse
+     * @returns {string} Text to use for the track line in the v-select menu
+     */
+    getTrackTitle(track: MediaStream): string {
+      if (track.DisplayTitle) {
+        return track.DisplayTitle;
+      }
+
+      return '';
+    },
+    /**
+     * @param {MediaStream} track - Track to parse
+     * @returns {string|undefined} Optional subtitle to use for the track line in the v-select menu
+     */
+    getTrackSubtitle(track: MediaStream): string | undefined {
+      if (
+        (this.type === 'Audio' || this.type === 'Subtitle') &&
+        track.Language
+      ) {
+        return this.getLanguageName(track.Language);
+      } else if (this.type === 'Audio' || this.type === 'Subtitle') {
+        return this.$t('undefined');
+      }
+
+      return undefined;
```

Comment:
This is unnecessary, as not returning anything will give you `undefined` anyway.